File: solutions/1389. Create Target Array in the Given Order/1389.cpp
// Time complexity: O(n^2)
// Space complexity: O(n)
class Solution {
 public:
  vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
    vector<int> ans;

    for (int i = 0; i < nums.size(); ++i)
      ans.emplace(ans.begin() + index[i], nums[i]);

    return ans;
  }
};

File: solutions/1389. Create Target Array in the Given Order/1389.py
# Time complexity: O(n^2)
# Space complexity: O(n)
class Solution:
  def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
    ans = []

    for num, i in zip(nums, index):
      ans.insert(i, num)

    return ans